core/lib/multivm/src/versions/vm_m5/transaction_data.rs

```diff
@@ -229,14 +229,13 @@ impl TransactionData {
 }
 
 pub(crate) fn derive_overhead(
-    gas_limit: u32,
+    gas_limit: u64,
     gas_price_per_pubdata: u32,
     encoded_len: usize,
 ) -> u32 {
-    assert!(
-        gas_limit <= MAX_TX_ERGS_LIMIT,
-        "gas limit is larger than the maximal one"
-    );
+    // Even if the gas limit is greater than the `MAX_TX_ERGS_LIMIT`, we assume that everything beyond `MAX_TX_ERGS_LIMIT`
+    // will be spent entirely on publishing bytecodes and so we derive the overhead solely based on the capped value
+    let gas_limit = std::cmp::min(MAX_TX_ERGS_LIMIT as u64, gas_limit);
```

Comment:
this is a bit more robust approach, considering that we want to allow proving u64 there. During block reexecution it will behave the same (in the past it was:

```
if x > MAX_TX_ERGS_LIMIT {
  panic
} else {
  x
}
```

now it is:

```
return min(x, MAX_TX_ERGS_LIMIT) => 

if x > MAX_TX_ERGS_LIMIT {
  //This should've never happened
} else {
  x
}
```

Reply:
Makes sense